### PR TITLE
Vulkan fixes from chromiumos-overlay

### PR DESCRIPTION
**Pull request by mattst88**
Some cross compiling fixes found in the chromiumos-overlay.

cc: @sarnex, @dm0-, @thesamesam, @SoapGentoo

### CODE REVIEW

File: media-libs/vulkan-loader/vulkan-loader-1.2.179.ebuild [hunk 1]
```diff
@@ -36,6 +35,10 @@ DEPEND="${PYTHON_DEPS}
 "
 PDEPEND="layers? ( media-libs/vulkan-layers:=[${MULTILIB_USEDEP}] )"
 
+PATCHES=(
+	"${FILESDIR}"/CHROMIUM-Fix-cross-compilation.patch
```
Comment (thesamesam) [inline media-libs/vulkan-loader/vulkan-loader-1.2.179.ebuild:39]: Missing?

File: media-libs/vulkan-layers/vulkan-layers-9999.ebuild [hunk 2]
```diff
@@ -46,9 +47,9 @@ multilib_src_configure() {
 		-DBUILD_WSI_XCB_SUPPORT=$(usex X)
 		-DBUILD_WSI_XLIB_SUPPORT=$(usex X)
 		-DBUILD_TESTS=OFF
-		-DGLSLANG_INSTALL_DIR="${EPREFIX}/usr"
-		-DCMAKE_INSTALL_INCLUDEDIR="${EPREFIX}/usr/include/vulkan/"
-		-DSPIRV_HEADERS_INSTALL_DIR="${EPREFIX}/usr/include/spirv"
+		-DGLSLANG_INSTALL_DIR="${ESYSROOT}/usr"
```
Comment (thesamesam) [inline media-libs/vulkan-layers/vulkan-layers-9999.ebuild:50]: This doesn’t seem right for any of these cases if the variables are genuinely used for installation — we never write to ESYSROOT, we just find headers there.

May need to check if the build system is abusing these variables to find stuff OR if these changes were just added with other changes and there was confusion about which had the needed effect.

Given this package is only going to install one thing, maybe the other variables are for finding? “spirv is installed here”, not “install spirv here”?

**Review by thesamesam — APPROVED** [review]

**Review by thesamesam — CHANGES_REQUESTED** [review]
(Didn’t mean to approve, on mobile)

File: media-libs/vulkan-loader/vulkan-loader-1.2.179.ebuild (hunk 1 above)
Comment (mattst88) [reply media-libs/vulkan-loader/vulkan-loader-1.2.179.ebuild:39]: Oops. Didn't mean to leave that hunk in.

File: media-libs/vulkan-layers/vulkan-layers-9999.ebuild (hunk 2 above)
Comment (mattst88) [reply media-libs/vulkan-layers/vulkan-layers-9999.ebuild:50]: I think these variables are for informing cmake where to find various dependencies, not for specifying locations to install files from this project. I'll double check.

Comment (mattst88) [reply media-libs/vulkan-layers/vulkan-layers-9999.ebuild:50]: Yes, I believe this is correct:

* `GLSLANG_INSTALL_DIR`: https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/CMakeLists.txt#L272
* `SPIRV_HEADERS_INSTALL_DIR`: https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/CMakeLists.txt#L268

I'm unsure about `CMAKE_INSTALL_INCLUDEDIR` though.

Comment (thesamesam) [reply media-libs/vulkan-layers/vulkan-layers-9999.ebuild:50]: Okay, fantastic, I've checked and I agree with those two. CMAKE_INSTALL_INCLUDEDIR seems like it should genuinely be ```${EPREFIX}``` though as it's only used for installing headers to the system:
```
install(FILES ${LAYER_UTIL_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/vulkan)
```

File: dev-util/vulkan-tools/vulkan-tools-9999.ebuild [hunk 3]
```diff
@@ -26,7 +26,7 @@ SLOT="0"
 IUSE="cube wayland +X"
 
 # Cube demo only supports one window system at a time
-REQUIRED_USE="!cube? ( || ( X wayland ) ) cube? ( ^^ ( X wayland ) )"
+REQUIRED_USE="cube? ( ^^ ( X wayland ) )"
```
Comment (thesamesam) [inline dev-util/vulkan-tools/vulkan-tools-9999.ebuild:30]: I think VULKAN_HEADERS_INSTALL_DIR and GLSLANG_INSTALL_DIR need to be ESYSROOT here as per other discussions (and consistency with the others)?

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: dev-util/vulkan-tools/vulkan-tools-9999.ebuild (hunk 3 above)
Comment (mattst88) [reply dev-util/vulkan-tools/vulkan-tools-9999.ebuild:30]: Indeed, fixed. I also noticed that `vulkan-loader` was previously fixed in commit fce28df7e8304ff4b93b1c194a1f8cb76876b571.

File: media-libs/vulkan-layers/vulkan-layers-9999.ebuild (hunk 2 above)
Comment (SoapGentoo) [reply media-libs/vulkan-layers/vulkan-layers-9999.ebuild:50]: I've rechecked all these variables, and they're fine now

**Review by SoapGentoo — APPROVED** [review]

**Comment by mattst88** [comment]
Merged. Thanks for the reviews!

**Pull request closed without merging**